### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

File: docs/IndexInvalidation.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ point of program execution.  A given index can be valid for zero, one or more
 than one collection instance at the same time.
 
 An index that is valid for a certain collection designates an element of that
-collection or represents a one-past-end index.
+collection or represents an one-past-end index.
```
Comment (gribozavr) [inline docs/IndexInvalidation.rst]: I'm not an expert in this, but I think it was correct before.

> You use an, not before a word beginning with vowels, but beginning with the sound of a vowel. 

https://www.quora.com/Which-is-correct-a-one-or-an-one

File: include/swift/Parse/ParserResult.h [hunk 2]
```diff
@@ -125,7 +125,7 @@ static inline ParserResult<T> makeParserCodeCompletionResult(T *Result =
 /// \brief Same as \c ParserResult, but just the status bits without the AST
 /// node.
 ///
-/// Useful when the AST node is returned by some other means (for example, a in
+/// Useful when the AST node is returned by some other means (for example, an in
```
Comment (gribozavr) [inline include/swift/Parse/ParserResult.h]: I don't think we need 'a' here at all.

**Comment by gribozavr** [comment]
@practicalswift LGTM otherwise.

**Comment by practicalswift** [comment]
@gribozavr Thanks fo reviewing! Fixed, squashed and pushed.

**Comment by practicalswift** [comment]
@gribozavr I'll remove the "a" you noticed in your comment ("I don't think we need 'a' here at all.") too. One second :-)

**Comment by practicalswift** [comment]
@gribozavr 

```/// Useful when the AST node is returned by some other means (for example, in
/// a vector out parameter).```

Could that be the correct placement of the "a"? :-)

**Comment by gribozavr** [comment]
@practicalswift I think that's correct :)

**Pull request merged**